### PR TITLE
Vendor `gardener@v1.52.2`

### PR DESCRIPTION
**Pull request by shafeeqes**
**How to categorize this PR?**

/area open-source
/kind enhancement

**What this PR does / why we need it**:
Vendor `gardener@v1.52.2`
https://github.com/gardener/gardener/releases/tag/v1.52.0
https://github.com/gardener/gardener/releases/tag/v1.52.1
https://github.com/gardener/gardener/releases/tag/v1.52.2
https://github.com/gardener/gardener/releases/tag/v1.51.0
https://github.com/gardener/gardener/releases/tag/v1.50.0

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
The following dependency is updated:
- github.com/gardener/gardener: v1.49.3 -> v1.52.2
- k8s.io/* : v0.23.3 -> v0.24.3
- sigs.k8s.io/controller-runtime v0.11.1 -> v0.12.1
- sigs.k8s.io/controller-tools: 0.8.0 -> 0.9.0
```

### CODE REVIEW

**Review by MartinWeindel — DISMISSED** [review]
/lgtm

Thanks for replacing logrus by logr!

**Review by MartinWeindel — APPROVED** [review]
/lgtm

File: test/system/shootdns_test.go [hunk 1]
```diff
@@ -113,7 +113,7 @@ func (f *shootDNSFramework) prepareClientsAndCluster() {
 }
 
 func (f *shootDNSFramework) createNamespace(ctx context.Context, namespace string) *v1.Namespace {
-	f.Logger.Printf("using namespace %s", namespace)
+	f.Logger.Info("using namespace %s", namespace)
```
Comment (ialidzhikov) [inline test/system/shootdns_test.go:116]: This actually logs the following:
```
{"level":"info","ts":"2022-08-15T16:44:21.173+0300","logger":"test","msg":"using namespace %s"}
```

**Pull request merged**